pkg/manager/member/dm_master_member_manager.go

```diff
@@ -730,19 +730,19 @@ func getNewMasterSetForDMCluster(dc *v1alpha1.DMCluster, cm *corev1.ConfigMap) (
 }
 
 func getMasterConfigMap(dc *v1alpha1.DMCluster) (*corev1.ConfigMap, error) {
-	config := &v1alpha1.MasterConfig{}
+	config := v1alpha1.NewMasterConfig()
 	if dc.Spec.Master.Config != nil {
 		config = dc.Spec.Master.Config.DeepCopy()
 	}
 
 	// override CA if tls enabled
 	if dc.IsTLSClusterEnabled() {
-		config.SSLCA = pointer.StringPtr(path.Join(dmMasterClusterCertPath, tlsSecretRootCAKey))
-		config.SSLCert = pointer.StringPtr(path.Join(dmMasterClusterCertPath, corev1.TLSCertKey))
-		config.SSLKey = pointer.StringPtr(path.Join(dmMasterClusterCertPath, corev1.TLSPrivateKeyKey))
+		config.Set("ssl-ca", path.Join(dmMasterClusterCertPath, tlsSecretRootCAKey))
+		config.Set("ssl-cert", path.Join(dmMasterClusterCertPath, corev1.TLSCertKey))
+		config.Set("ssl-key", path.Join(dmMasterClusterCertPath, corev1.TLSPrivateKeyKey))
 	}
 
-	confText, err := MarshalTOML(config)
+	confText, err := config.MarshalTOML()
```

Comment:
We need to make sure no rolling update is triggered for the existing DM cluster with configuration set explicitly after upgrading the TiDB Operator.
Please refer to https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tikv_member_manager.go#L248.

Reply:
dm has used mngerutils.UpdateConfigMapIfNeed() too